rust/src/table_state.rs

```diff
@@ -330,6 +345,39 @@ impl DeltaTableState {
 
         Ok(())
     }
+
+    /// Obtain Add actions for files that match the filter
+    pub fn get_active_add_actions_by_partitions<'a>(
```

Comment:
Moving this function to the state was part of an experiment. I did not move it back, since it exclusiveliy works on the state and from what I can see we are going to need functionality as such as we are evolving operations capabilities.